### ISSUE TITLE
Configurable ConstantName Check

### ISSUE THREAD

**Issue by mishadoff** [checkstyle/checkstyle#1106]
Variable names which are skipped by `ConstantName` check are hardcoded 

https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java#92

Allow to configure set of variable names to skip using `ConstantName` module in `checkstyle.xml`

The real use case for it is to skip on name `logger`.

As a workaround we use `@SuppressWarnings`

**Comment by romani** [comment]
by "token" you mean variable names ?
Example: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java#L92

**Comment by mishadoff** [comment]
Yes.

**Comment by romani** [comment]
new option `ignoredNames` have to be created that will store list of variable names separated by ","

**Comment by mkordas** [comment]
Shouldn't we enable users to provide ignoredNames also as regexp?

**Comment by romani** [comment]
@mkordas , yes that is good idea.

**Comment by Vladlis** [comment]
I'll do this one

**Comment by mkordas** [comment]
@Vladlis - that's great. I think the most common example that we can put to docs would be `log(ger)?`.

**Comment by Vladlis** [comment]
@mishadoff, why not just use property 'format' ?

```bash
vlad@vlad-N61PC-M2S ~/cli $ java -jar checkstyle-6.8.1-all.jar -c my_check.xml Test.java
Starting audit...
Audit done.
vlad@vlad-N61PC-M2S ~/cli $ cat Test.java
package com.puppycrawl.tools.checkstyle.naming;

import java.io.ObjectStreamField;
import java.util.logging.Logger;

public class InputConstantNames
{
    private static final long serialVersionUID = 1L; //should be ignored
    private static final ObjectStreamField[] serialPersistentFields = {}; // should be ignored too
    private static final Logger log = Logger.getLogger("");
    private static final Logger logger = Logger.getLogger("");
}

vlad@vlad-N61PC-M2S ~/cli $ cat my_check.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">


<module name = "Checker">
    <module name="TreeWalker">
         <module name="ConstantNameCheck">
			<property name="format" value="^log(ger)?|[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"/>
		 </module>
    </module>
</module>
```

**Comment by mishadoff** [comment]
@Vladlis this is could be an option, any drawbacks and misses?

**Comment by Vladlis** [comment]
@mishadoff, I don't see any drawbacks. This example just extends default pattern for the Check.
@romani, @mkordas, what do you think?

**Comment by romani** [comment]
@Vladlis , good catch! no need to add another option.

Please add "ConstantName" to http://checkstyle.sourceforge.net/config_naming.html#Enhancements with such example as it is not obvious :) even for owners of project.

**Comment by romani** [comment]
merged
